### ISSUE TITLE
zephyrbot should post a new message instead of editing the old one

### ISSUE THREAD

**Issue by ulfalizer** [zephyrproject-rtos/ci-tools#61]
I've seen people be confused by the current behavior a few times now, and it gets really bad when the zephyrbot message is hidden behind a "x more messages" that you have to click.

The old message should probably be deleted as well.

**Comment by marc-hb** [comment]
A closely related issue: all results seem to be deleted once the PR is merged. See example at
https://github.com/zephyrproject-rtos/zephyr/pull/17029

> The old message should probably be deleted as well.

Not sure about this. Hiding test results a bit to avoid clutter sure why not, but deleting  them no, rarely ever. For shippable it would be even worse because it would only delete the pointer to them without even freeing up anything on Shippable, it would just make them extremely difficult to find. I've been working around this by copying Shippable URLs into github comments to "rescue" them.

**Comment by aescolar** [comment]
>The old message should probably be deleted as well.

Editing the old message means you can still see the old in the history. Which can be useful in some cases.
I actually like the fact that it edits the old message instead of generating a new one, it is cleaner, and it clearly indicates the old message is not valid anymore.
Another thing is that the zephyr bot message tends to be the first or one of the first in the PR, so it is easy to find.

**Comment by aescolar** [comment]
> I've been working around this by copying Shippable URLs into github comments to "rescue" them

@marc-hb The last CI result is still linked after a PR is merged. You can find it by clicking in the "View Details" button in the merged line (the one with the violet merge icon). (If you meant results from runs before the last CI run, that is another story)

**Comment by marc-hb** [comment]
> @marc-hb The last CI result is still linked after a PR is merged. You can find it by clicking in the "View Details" button in the merged line

... except for CI failures, which means it's useless? Example:
https://github.com/zephyrproject-rtos/zephyr/pull/17029

**Comment by nashif** [comment]
> A closely related issue: all results seem to be deleted once the PR is merged. See > example at zephyrproject-rtos/zephyr#17029

Not sure how this is related. This is a screwup in how Github deals with suggested changes, has really nothing to do with our CI AFAIK.

Fully agree with @aescolar, I thin we should keep it the same way. Having the bot go wild posting new messages will just make the PRs not easy to read and parse. The diff of the bot message is very useful, it actually tells you what has been fixed since last time.

closing as wont-fix.

**Comment by marc-hb** [comment]
> Not sure how this is related

I didn't fully realize yet that the same CI failures are reported in two different and often distant places:
- zephyrbot's message
- [github's status checks](https://help.github.com/en/articles/about-required-status-checks)

Not counting the "checks" tab which is yet another (and unused) third thing.

If the "details" buttons in the status checks linked to the failure logs (as they already do for Shippable) then maybe the zephyrbot message wouldn't be necessary. Except force pushes remove that part of the history too?
